packages/devtools_testing/lib/inspector_service_test.dart

```diff
@@ -83,12 +82,194 @@ Future<void> runInspectorServiceTests(FlutterTestEnvironment env) async {
+        expect(rootDirectories.length, 1);
+        expect(rootDirectories.first, endsWith('/fixtures/flutter_app'));
         await group.dispose();
       });
 
+      test('localClasses', () async {
+        await env.setupEnvironment();
+        final group = inspectorService.createObjectGroup('test-group');
+        // These tests are moot if widget creation is not tracked.
+        expect(await inspectorService.isWidgetCreationTracked(), isTrue);
+        await inspectorService.setPubRootDirectories([]);
+        final List<String> rootDirectories =
+            await inspectorService.inferPubRootDirectoryIfNeeded();
+        expect(rootDirectories.length, 1);
+        expect(rootDirectories.first, endsWith('/fixtures/flutter_app'));
+        final originalRootDirectories = rootDirectories.toList();
+        try {
+          expect(
+            (inspectorService.localClasses.keys.toList()..sort()),
+            equals(
+              ['MyApp', 'MyOtherWidget', 'NotAWidget'],
+            ),
+          );
+
+          await inspectorService
+              .setPubRootDirectories(['${rootDirectories.first}/lib/src']);
+          // Adding src does not change the directory as local classes are
+          // computed at the library level.
+          expect(
+            (inspectorService.localClasses.keys.toList()..sort()),
+            equals(
+              ['MyApp', 'MyOtherWidget', 'NotAWidget'],
+            ),
+          );
+
+          expect(
+              inspectorService.rootPackages.toList(), equals(['flutter_app']));
+          expect(inspectorService.rootPackagePrefixes.toList(), isEmpty);
+
+          await inspectorService.setPubRootDirectories(
+              ['/usr/jacobr/foo/lib', '/usr/jacobr/bar/lib/bla']);
+          expect(
+              inspectorService.rootPackages.toList(), equals(['foo', 'bar']));
+          expect(inspectorService.rootPackagePrefixes.toList(), isEmpty);
+
+          expect(
+            inspectorService.isLocalUri('package:foo/src/bar.dart'),
+            isTrue,
+          );
+          expect(
+            inspectorService.isLocalUri('package:foo.bla/src/bar.dart'),
+            isFalse,
+          );
+          expect(
+            inspectorService.isLocalUri('package:foos/src/bar.dart'),
+            isFalse,
+          );
+          expect(
+            inspectorService.isLocalUri('package:bar/src/bar.dart'),
+            isTrue,
+          );
+          expect(
+            inspectorService.isLocalUri(
+              'package:bar.core/src/bar.dart',
+            ),
+            isFalse,
+          );
+          expect(
+            inspectorService.isLocalUri(
+              'package:bar.core.bla/src/bar.dart',
+            ),
+            isFalse,
+          );
+          expect(
+            inspectorService.isLocalUri(
+              'package:bar.cores/src/bar.dart',
+            ),
+            isFalse,
+          );
+
+          // Test bazel project like cases.
```

Comment:
can this be put in a separate test case? `test('local classes for bazel projects', () {...});`

Reply:
done